### DISCUSSION TITLE
Manual Microsoft Authentication

### DISCUSSION THREAD

**Discussion by DaniEzzeddine** [O365/python-o365#629] [Q&A]
Hi everyone,

I'm a bit stuck and trying to figure out the right way to use the lib.

I need to handle token creation for different o365 accounts from 1 browser in order to send emails from different email addresses. So far when I set `account.new_message(resource=args.sender_email)`  - I would get an error message that credentials invalid since the authentication where done from a different email address. So I have to manually log out of Microsoft account and log in with the desired email. Does anybody know how to log in to Microsoft from o365 lib authentication?

**Comment by alejcas** [comment]
If you are using the default auth method then you will need to give consent to each account and store the different tokens in different places. Then use one token or another to send email as each account.

if you use the credentials auth method, the you are logged as the “app” and therefore you will have to set the appropriate permissions for the different accounts in order to send email from each one.

**Reply by DaniEzzeddine** [reply]
Thank you, I figured that part. I'm having trouble with - using the default auth method, is it possible to trigger authentication to Microsoft account every time I'm trying to retrieve a new token, cause currently I would get redirected to log in page, I would automatically get logged in inside Microsoft without the ability to choose an account I want to use. So I just need to make user log in with desired email to Microsoft account every time his token is expired.

**Reply by alejcas** [reply]
But just add the offline_access permission and you will have (if you take care) no expiring tokens (aka long lived sessions).
